src/Entity/Character.cpp

```diff
@@ -54,56 +64,49 @@ void Character::CharacterUpdate(flecs::entity characterEntity, TransformComponen
 		newPosition.y >= 0 && newPosition.y <= mapTexture.height-CHARACTER_HEIGHT)
 	{
-		const Rectangle newCharacterRect {newPosition.x, newPosition.y, collisionComponent.mRectScale.x, collisionComponent.mRectScale.y};
-
+		const Rectangle newCharacterRect{newPosition.x, newPosition.y, collisionComponent.mRectScale.x, collisionComponent.mRectScale.y};
 		bool hasCollided{false};
-		//auto q = ecs.query<CollisionComponent, TransformComponent>();
-		//q.each([&hasCollided,&newCharacterRect, &characterEntity](flecs::entity e, CollisionComponent& otherCollisionComponent, TransformComponent& otherTransformComponent){
-		//	// Avoid computing a collision detection if we already detected a collision.
-		//	if (hasCollided)
-		//		return;
-		//	
-		//	const Rectangle otherRect {otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherCollisionComponent.mRectScale.x, otherCollisionComponent.mRectScale.y};
-		//	if (e != characterEntity && CheckCollisionRecs(newCharacterRect, otherRect))
-		//	{
-		//		hasCollided = true;
-		//	}
-		//});
-
-		//QUESTION: I couldnt pass two components into .each query. I tried to create query and use it, it leads to a crash. Apparently I cant create a query inside a system? 
-		//passing a query from outside seems really complicated? Is there a better way to do each with entities that should have several components?
-		//atm I just use transform scale, since it is the same (for rectangle collision) as collision component would be. 
-
-		characterEntity.world().each<TransformComponent>([&hasCollided, &newCharacterRect, &characterEntity](flecs::entity e, TransformComponent& otherTransformComponent) {
+		Rectangle otherRect{};
+		auto filter = characterEntity.world().filter<CollisionComponent, TransformComponent>();
+
+		filter.each([&hasCollided, &newCharacterRect, &otherRect, &characterEntity](flecs::entity e, CollisionComponent& otherCollisionComponent, TransformComponent& otherTransformComponent) {
+			// Avoid computing a collision detection if we already detected a collision.
 			if (hasCollided)
 				return;
-			const Rectangle otherRect{otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherTransformComponent.mScale.x, otherTransformComponent.mScale.y};
+
+			otherRect = {otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherCollisionComponent.mRectScale.x, otherCollisionComponent.mRectScale.y};
 			if (e != characterEntity && CheckCollisionRecs(newCharacterRect, otherRect))
 			{
 				hasCollided = true;
 			}
 		});
 
-
 		if (!hasCollided)
 		{
 			transformComponent.mPosition = newPosition;
 		}
 	}
 }
 
-}// namespace game
\ No newline at end of file
+}// namespace game
+
+//QUESTION: I have two ways of doing collisions now, one with .each and one with .iter. The .iter
+// advantage is that i can break out of the loop and dont have to go through each entity if i collide on some
+//of it earlier?
```

Comment:
Indeed, in this particular case, `.iter` will be better since you're given all the collision components at once.